huaweicloud/services/css/data_source_huaweicloud_css_flavors.go

```diff
@@ -0,0 +1,202 @@
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"ess", "ess-cold", "ess-master", "ess-client"}, false),
+			},
+			"version": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"memory": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"vcpus": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"flavors": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"region": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"version": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"memory": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"vcpus": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"disk_range": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceCssFlavorsRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+	client, err := config.CssV1Client(region)
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating CSS V1 client: %s", err)
+	}
+
+	flavorsResp, err := cluster.ListFlavors(client)
+	if err != nil {
+		return fmtp.DiagErrorf("Unable to retrieve CSS flavors: %s ", err)
+	}
+
+	allFlavors := flatternFlavors(flavorsResp)
+
+	if len(allFlavors) < 1 {
+		return fmtp.DiagErrorf("No data found. Please change your search criteria and try again.")
+	}
+
+	filter := map[string]interface{}{
+		"Region":  region,
+		"Type":    d.Get("type"),
+		"Name":    d.Get("name"),
+		"Version": d.Get("version"),
+	}
+
+	if v, ok := d.GetOk("cpu"); ok {
+		filter["Cpu"] = v
+	}
+
+	if v, ok := d.GetOk("ram"); ok {
+		filter["Ram"] = v
+	}
+
+	filterFlavors, err := utils.FilterSliceWithField(allFlavors, filter)
+	if err != nil {
+		return fmtp.DiagErrorf("filter CSS flavors failed: %s", err)
+	}
+	logp.Printf("filter %d CSS flavors from %d through options %v", len(filterFlavors), len(allFlavors), filter)
+
+	if len(filterFlavors) < 1 {
+		return fmtp.DiagErrorf("No data found. Please change your search criteria and try again.")
+	}
+
+	mErr := d.Set("flavors", buildFlavors(filterFlavors))
+	if mErr != nil {
+		return fmtp.DiagErrorf("set flavors err:%s", mErr)
+	}
+
+	uuid, err := uuid.GenerateUUID()
```

Comment:
suggest use `resource.UniqueId()` , more reliable

Reply:
The R015 analyzer reports (*schema.ResourceData).SetId() usage with **unstable** resource.UniqueId() value. Schema attributes should be stable across Terraform runs.

keep it as it is.